lib/phobos/listener.rb

```diff
@@ -9,9 +9,28 @@ class Listener
     DEFAULT_MAX_BYTES_PER_PARTITION = 1_048_576 # 1 MB
     DELIVERY_OPTS = %w[batch message inline_batch].freeze
 
-    attr_reader :group_id, :topic, :id
-    attr_reader :handler_class, :encoding, :consumer
-
+    # @return [String]
+    attr_reader :group_id
+    # @return [String]
+    attr_reader :topic
+    attr_reader :id
+    # @return [Class]
+    attr_reader :handler_class
+    attr_reader :encoding, :consumer
+
+    # @param handler [Class]
+    # @param group_id [String]
+    # @param topic [String]
+    # @param min_bytes [Integer]
+    # @param max_wait_time [Integer]
+    # @param start_from_beginning [Boolean]
+    # @param delivery [String]
+    # @param max_bytes_per_partition [Integer]
```

Comment:
Just curious - are all these annotations automatically generated? How come some of the parameters are missing? e.g. force_encoding, backoff

Reply:
Nope, they're done by hand, and that's why some are missing. :) Most of the missing info comes from RubyKafka. I didn't actually want to document every type here - I'm more interested in the method definitions themselves (i.e. X object responds to Y method that takes Z parameters) so that I can use them to document Deimos. Once that's done I could move back to this and fill in whatever's missing.